### ISSUE TITLE
Special case movement | Expediting cases

### ISSUE THREAD

**Issue by pshahVA** [department-of-veterans-affairs/caseflow#11397]
# Context

Caseflow built critical functionality for AMA cases to be processed in time for the law’s implementation. The team focused on the most straightforward path (also known as the happy path) an appeal could take to meet this deadline, and get to a “minimum viable” state.

Caseflow is now iterating on this initial feature set to handle other paths an appeal could take.

# Goals for this functionality:

- Allow the Board to complete daily case movement tasks without support/help desk intervention
- Allow the Board to expedite AMA appeals when necessary

# Strategy:

- Identify the gaps in business needs versus product functionality
- Use a phased approach to bring these enhancements to production. We’ve proposed which phases 
   to go first based on (1) the support tickets we’ve already gotten and (2) where the majority of AMA 
   cases are waiting
- Deploy, Review, Iterate (Pilot if needed)

# Proposed agile implementation:

1.	Identify gaps (Complete)
2.	Allow judges to reassign their AMA tasks (Complete 7/4/19)
3.	Allow chosen users to manually distribute AMA cases, that are ready for distribution and have no 
        pending child tasks, to judges for decision (Complete 7/29/2019)
4.	Allow chosen users to manually distribute AMA hearing docket cases to judges for decision
5.	Allow users to manually distribute AMA cases with blocking tasks (e.g. FOIA requests, Evidence Window)

# Phase 1 

Identify Gaps:
•	Initial review of gaps is complete
•	As we develop and iterate, more requirements and scenarios could be discovered

# Phase 2:

Allow judges to reassign their AMA tasks:

•	This is complete as of 7/4/19
•	Judges can assign any of their tasks assigned to them to other judges
Note: This helps with acting VLJs and SSCs reassigning their own judge tasks

# Phase 3:

Distribute a case with no child tasks:

- This is complete as pf 7/29/2019
- Bypass automatic case distribution algorithm for special cases. Initially, only for appeals with no blocking tasks or administrative actions (e.g. FOIA requests, Evidence Windows)
- Justification: We propose doing this first because (1) it encompasses the majority of the support tickets we’ve already gotten and (2) waiting for distribution is one of the places where the majority of AMA cases are waiting

- This will create and immediately complete a task so this movement is tracked and reflected in case timeline and for reporting purposes. 
- Question: We propose the task be named “Extraordinary case movement” but ultimately this is up to the Board. What should this task be called?

- In this phase, we will also create an organization/team that contains users who are authorized to do this movement. 
- Question: The Board can add users to this organization as it sees fit. This organization needs a name. One proposal could be “Case movement”. What should this organization be called?

# Phase 4

Expedite hearing docket cases:

●	Allow hearing docket cases, that aren’t already ready for distribution to a judge, to be distributed
●	Allow chosen users to cancel, withdraw, or bypass hearing tasks, such as Evidence Window tasks 
        or Transcription tasks, so the case can be distributed to a judge
●	This, too, would create and immediately complete the “extraordinary” task
●	Justification: There is a high volume of AMA hearing docket cases, so we think this scenario could 
        happen frequently.

# Phase 5

Distribute cases with blocking child tasks
●	Users are prompted to recognize, and if appropriate, act on blocking child tasks so the case can 
        be distributed to a judge
○	For example, admin actions can be cancelled or reassigned to appropriate teams

●	This, too, would create and immediately complete the “extraordinary” task

---- 
Moved out of scope of this ticket, into #11801 

Phase 5 

Allow chosen users to act on behalf of judges

•	Allow chosen users to reassign judge tasks to other judges to account for judges leave, sickness 
        etc.

Phase 6

Chosen users can act on behalf of DVCs
•	Allow chosen users to act on behalf of DVCs, to include:
•	Assigning their tasks to other judges
•	Requesting cases
•	This encompasses DVC Staff Assistants working on behalf of DVCs

**Comment by pshahVA** [comment]
@lomky @laurjpeterson @nicholer  Added this epic

**Comment by lomky** [comment]
https://docs.google.com/document/d/1W84vBfRUTVuEDzWz_GR8NQRKaw7waNn8w419AE5PIQc/edit#

**Comment by lomky** [comment]
https://dsva.slack.com/archives/CJL810329/p1561119080058400?thread_ts=1561119015.058300

**Comment by stwards11** [comment]
How many scenarios do we account for in the initial release?

**Comment by stwards11** [comment]
Do we set overarching rules for special case movement that automatically when a "special case movement task" (as in step #3) is created that all transcription related tasks be unblocked?

**Comment by stwards11** [comment]
User story: As a user I need the ability to first acknowledge specific hearing related tasks and then have the ability to bypass and override them so that i can move a case to a judge quickly

**Comment by stwards11** [comment]
Would this qualify as a scenario for special case movement?

"Can someone cancel the attorney draft task on this case. Put it back in the VLJs 'assign' queue. The case has some issues with 'issues' and shouldn't be assigned to any attorney at the moment while the board figures stuff out."

created a new JudgeAssignTask `318322` and assigned to `BVASSORATHIA` and cancelled JudgeDecisionReviewTask `315117`.

Would this or any previous workarounds created qualify as special case movement??

**Comment by lomky** [comment]
I wouldn't qualify that as special case movement, personally. I think that could just be a new action available to a VLJ on their cases (pull back / cancel assignment to attorney)

**Comment by stwards11** [comment]
Scenarios for special case movement: 
1). Veteran becomes terminally ill
2). Veteran has a hardship situation (Does not have food or shelter, medicine, etc. and as a result health and safety are jeopardized

**Comment by lomky** [comment]
@stwards11 aren't those AOD scenarios?

**Comment by stwards11** [comment]
@lomky: Upon further research you are correct.

**Comment by stwards11** [comment]
Scenario:  A veteran has a hearing scheduled and has not waived the 90 day evidence window and is currently in day 35.  A judge requests to have this case pulled for review immediately.  

1). A judge would need to enable "special case movement"
2). Select the specific case to be moved
3). He would then be prompted that the evidence window if open will automatically be closed: (Modal) "Some sort of notification that by committing this action the evidence window will be closing"
     *If special case movement is selected transcription windows will automatically stay open

**Comment by stwards11** [comment]
Can expediting cases/special case movement be defined as:  A specific user or set of users that are granted rights to bypass, cancel, or withdraw specific hearing related functions for hearing docket cases, that aren’t already ready for distribution to a judge, to be distributed?

**Comment by laurjpeterson** [comment]
@stwards11 - based on our brief convo the other day, I agree that administratively acting on behalf of other users could be split out into another effort. Here's the ticket I created: https://github.com/department-of-veterans-affairs/caseflow/issues/11801

**Comment by laurjpeterson** [comment]
> Can expediting cases/special case movement be defined as: A specific user or set of users that are granted rights to bypass, cancel, or withdraw specific hearing related functions for hearing docket cases, that aren’t already ready for distribution to a judge, to be distributed?

Yes

**Comment by stwards11** [comment]
User scenarios to be considered:

Scenario # 1 - A hearing has been held and the evidence window is open
Scenario # 2 - A veteran has scheduled a hearing
Scenario # 3 - A veteran has postponed a hearing
Scenario # 4 - A veteran does not have a hearing scheduled

**Comment by laurjpeterson** [comment]
Reviewed phase 4 - expediting hearing cases - with Carolyn today. next steps:
- create low-fidelity mock ups of each scenario, with their relevant tasks - Shawn
- write content to explain what the user is doing for each scenario - Shawn

Paper prototypes until Shawn gets Mural access

**Comment by laurjpeterson** [comment]
Posting w/r/t blocking hearing docket appeal tasks. Certain users (I assume hearing coordinators?) can access the Hearing Details page and check the box "Waive 90 day evidence window", which, I assume, cancels the Evidence Window task

<img width="1105" alt="Screen Shot 2019-11-06 at 3 05 04 PM" src="https://user-images.githubusercontent.com/27704093/68333623-de28ac00-00a6-11ea-980e-f33dd8c07d9c.png">

**Comment by lomky** [comment]
Generally speaking yes, @laurjpeterson. Specifically, that checkbox will prevent the creation of the Evidence Window task altogether. The Transcription task is created regardless, though, which still blocks distribution.

**Comment by jimruggiero** [comment]
@hschallhorn It appears that all work on this epic is complete. Can you confirm? If yes, can/should this epic be closed and a new epic started when we hear back from JW and JG at BVA?

**Comment by hschallhorn** [comment]
Phase 4 and 5 are not complete, they  need to have tickets written and prioritized/worked. We do know that JW specifically said they wanted Phase 4.

**Comment by araposo-tistatech** [comment]
Note, phases 4 & 5 combined for implementation.
